File: sully/c/Sully.c
#include <stdio.h>
#include <fcntl.h>
#include <string.h>
#include <stdlib.h>
#include <unistd.h>

int	main()
{
	int i = 5;
	if (access("Sully_5.c", F_OK) != -1)
		--i;

	if (i >= 0) {
		char file_name[16] = "Sully_";
		file_name[6] = i + '0';
		
		char file_path[32] = {0};
		strcat(file_path, file_name);
		strcat(file_path, ".c");

		FILE *fptr = fopen(file_path, "w");
		if (!fptr)
			return 1;

		char source_code[] = "#include <stdio.h>%3$c#include <fcntl.h>%3$c#include <string.h>%3$c#include <stdlib.h>%3$c#include <unistd.h>%3$c%3$cint%4$cmain()%3$c{%3$c%4$cint i = %2$d;%3$c%4$cif (access(%5$cSully_5.c%5$c, F_OK) != -1)%3$c%4$c%4$c--i;%3$c%3$c%4$cif (i >= 0) {%3$c%4$c%4$cchar file_name[16] = %5$cSully_%5$c;%3$c%4$c%4$cfile_name[6] = i + '0';%3$c%4$c%3$c%4$c%4$cchar file_path[32] = {0};%3$c%4$c%4$cstrcat(file_path, file_name);%3$c%4$c%4$cstrcat(file_path, %5$c.c%5$c);%3$c%3$c%4$c%4$cFILE *fptr = fopen(file_path, %5$cw%5$c);%3$c%4$c%4$cif (!fptr)%3$c%4$c%4$c%4$creturn 1;%3$c%3$c%4$c%4$cchar source_code[] = %5$c%1$s%5$c;%3$c%4$c%4$cchar input[4096];%3$c%4$c%4$csprintf(input, source_code, source_code, i, 10, 9, 34);%3$c%4$c%4$cfprintf(fptr, %5$c%%s%5$c, input);%3$c%4$c%4$cfclose(fptr);%3$c%3$c%4$c%4$cchar compile_cmd[128] = %5$ccc -Wall -Wextra -Werror %5$c;%3$c%4$c%4$cstrcat(compile_cmd, file_name);%3$c%4$c%4$cstrcat(compile_cmd, %5$c.c%5$c);%3$c%4$c%4$cstrcat(compile_cmd, %5$c -o %5$c);%3$c%4$c%4$cstrcat(compile_cmd, file_name);%3$c%4$c%4$csystem(compile_cmd);%3$c%3$c%4$c%4$cif (i > 0) {%3$c%4$c%4$c%4$cchar exec_cmd[128] = %5$c./%5$c;%3$c%4$c%4$c%4$cstrcat(exec_cmd, file_name);%3$c%4$c%4$c%4$csystem(exec_cmd);%3$c%4$c%4$c}%3$c%4$c}%3$c%4$creturn 0;%3$c}";
		char input[4096];
		sprintf(input, source_code, source_code, i, 10, 9, 34);
		fprintf(fptr, "%s", input);
		fclose(fptr);

		char compile_cmd[128] = "cc -Wall -Wextra -Werror ";
		strcat(compile_cmd, file_name);
		strcat(compile_cmd, ".c");
		strcat(compile_cmd, " -o ");
		strcat(compile_cmd, file_name);
		system(compile_cmd);

		if (i > 0) {
			char exec_cmd[128] = "./";
			strcat(exec_cmd, file_name);
			system(exec_cmd);
		}
	}
	return 0;
}

File: sully/go/Sully.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"os/exec"
	"strconv"
)

func main() {
	i := 5

	if _, err := os.Stat("Sully_5.go"); err == nil {
		i -= 1
	}

	if i >= 0 {
		file_name := "Sully_" + strconv.Itoa(i)
		file_path := file_name + ".go"

		fptr, err := os.OpenFile(file_path, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0755)
		if err != nil {
			os.Exit(1)
		}

		s := "package main%[3]c%[3]cimport (%[3]c%[4]c%[5]cbufio%[5]c%[3]c%[4]c%[5]cfmt%[5]c%[3]c%[4]c%[5]cos%[5]c%[3]c%[4]c%[5]cos/exec%[5]c%[3]c%[4]c%[5]cstrconv%[5]c%[3]c)%[3]c%[3]cfunc main() {%[3]c%[4]ci := %[2]d%[3]c%[3]c%[4]cif _, err := os.Stat(%[5]cSully_5.go%[5]c); err == nil {%[3]c%[4]c%[4]ci -= 1%[3]c%[4]c}%[3]c%[3]c%[4]cif i >= 0 {%[3]c%[4]c%[4]cfile_name := %[5]cSully_%[5]c + strconv.Itoa(i)%[3]c%[4]c%[4]cfile_path := file_name + %[5]c.go%[5]c%[3]c%[3]c%[4]c%[4]cfptr, err := os.OpenFile(file_path, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0755)%[3]c%[4]c%[4]cif err != nil {%[3]c%[4]c%[4]c%[4]cos.Exit(1)%[3]c%[4]c%[4]c}%[3]c%[3]c%[4]c%[4]cs := %[5]c%[1]s%[5]c%[3]c%[4]c%[4]cw := bufio.NewWriter(fptr)%[3]c%[4]c%[4]cfmt.Fprintf(w, s, s, i, 10, 9, 34)%[3]c%[4]c%[4]cw.Flush()%[3]c%[3]c%[4]c%[4]cif err := fptr.Close(); err != nil {%[3]c%[4]c%[4]c%[4]cos.Exit(1)%[3]c%[4]c%[4]c}%[3]c%[3]c%[4]c%[4]ccompile_cmd := exec.Command(%[5]c/bin/sh%[5]c, %[5]c-c%[5]c, %[5]cgo build -o %[5]c+file_name+%[5]c %[5]c+file_path)%[3]c%[4]c%[4]ccompile_cmd.Start()%[3]c%[4]c%[4]ccompile_cmd.Wait()%[3]c%[3]c%[4]c%[4]cif i > 0 {%[3]c%[4]c%[4]c%[4]cexec_cmd := exec.Command(%[5]c./%[5]c + file_name)%[3]c%[4]c%[4]c%[4]cexec_cmd.Start()%[3]c%[4]c%[4]c%[4]cexec_cmd.Wait()%[3]c%[4]c%[4]c}%[3]c%[4]c}%[3]c}%[3]c"
		w := bufio.NewWriter(fptr)
		fmt.Fprintf(w, s, s, i, 10, 9, 34)
		w.Flush()

		if err := fptr.Close(); err != nil {
			os.Exit(1)
		}

		compile_cmd := exec.Command("/bin/sh", "-c", "go build -o "+file_name+" "+file_path)
		compile_cmd.Start()
		compile_cmd.Wait()

		if i > 0 {
			exec_cmd := exec.Command("./" + file_name)
			exec_cmd.Start()
			exec_cmd.Wait()
		}
	}
}
